### ISSUE TITLE
More Python best practises

### ISSUE THREAD

**Issue by robinandeer** [robinandeer/software-development#4]
Steal stuff from @guillermo-carrasco:

https://github.com/guillermo-carrasco/BestPracticesWorkshop/blob/master/DOCS/styling.md

**Comment by robinandeer** [comment]
Well at least I started... :dizzy_face:

This is a lot more comprehensive than I first imagined :smile: (even if some examples are just copied from the official PEP8 proposal).

**Comment by guillermo-carrasco** [comment]
Most of them are! Why to change something that is good :-P

**Comment by robinandeer** [comment]
Well I guess it's just my obsession with the DRY principle, hehe :sweat_smile: 
Copy-pasting content when I feel like I could link to the original source doesn't come easy.

**Comment by guillermo-carrasco** [comment]
That is very true :-)

Is just that I wanted a reduce subset of PEP8, not the whole thing, basically for two reasons: 
1. I don't completely agree with the whole PEP8
2. For the aim of that course, I thought it was better to have those styling "rules" summarised with the rest of the course material

kudos for this repo!

**Comment by robinandeer** [comment]
Thank you!

It did make more sense in context of the course to aggregate the PEP8 subset in a single document, I agree. For this project it would be better to accept PEP8 completely with some clarifications I think.

If you have time I would be interested in what points you disagree with PEP8?

**Comment by guillermo-carrasco** [comment]
hmm okay I'm going to be honest here: I may have read PEP8 rule too long ago, because I read them back now (to answer your question) and... well basically I agree on everything. Or at least it is how _I think_ I code so... nah, forget my first point! (the second one still holds though)

**Comment by robinandeer** [comment]
Memories of the course are starting to come back to me... For example @vals talking about how he used to go over his code replacing all list comprehensions with generator comprehensions. That made a big mark on me.

BTW: can I find this document, mentioned in the README, somewhere?
"Development workflow at Science For Life Laboratory"

**Comment by guillermo-carrasco** [comment]
Yeah I just pushed a commit fixing the broken link link. [Here](https://github.com/guillermo-carrasco/BestPracticesWorkshop#development-workflow-at-science-for-life-laboratory) you go
